Add release highlights

anyasabo opened the pull request:
First stab at release highlights. Edits veeeeeeeery welcome. Release notes are here:
https://www.elastic.co/guide/en/cloud-on-k8s/1.0/release-notes-1.0.0.html
Which is what I used to cultivate this. I wonder if it's worth calling out all the various upgrade resiliency changes we made?

We will still want to flesh out the upgrading ECK doc when we decide on this
https://github.com/elastic/cloud-on-k8s/issues/2308

pebrc commented on docs/release-notes/highlights-1.0.0.asciidoc:
```diff
@@ -0,0 +1,32 @@
+[id="{p}-release-license-mgmt"]
+=== License management
+
+Handling of licenses has been greatly improved. If the license changes, all new clusters will use it and all existing clusters will be upgraded to the new cluster. See <<{p}-licensing>> for more information.
```
```suggestion
Handling of licenses has been greatly improved. If the license changes, all new clusters will use it and all existing clusters will be upgraded to the new license. See <<{p}-licensing>> for more information.
```

anyasabo replied:
😄 good catch

anyasabo commented:
Went through unlabeled PRs and added a v1.0.0 label and the type label where appropriate, then regenerated release notes. Also added the istio bit as discussed in slack.

thbkrkr commented:
Can you regenerate the release notes once again please, as some PR titles have been adjusted (#2385)?

sebgl commented on docs/release-notes/highlights-1.0.0.asciidoc:
```diff
@@ -0,0 +1,38 @@
+[id="{p}-release-v1"]
+=== Resource version promotion
+
+Custom resources have graduated to version `v1`.  `v1beta1` manifests will still function, but are deprecated and should be updated to `v1`. Support for `v1beta1` will be removed in a future release. See <<{p}-upgrading-eck>> for more information.
```
Maybe worth mentioning that support for `v1alpha1` resources is removed.

alaudazzi commented on docs/release-notes/1.0.0.asciidoc:
```diff
@@ -22,27 +22,42 @@
 
+* Do not report max ERUs for basic licenses {pull}2377[#2377]
+* Fix v1beta1 webhook {pull}2358[#2358] (issue: {issue}2355[#2355])
+* Preserve environment variable order in pod spec {pull}2341[#2341]
```
```suggestion
* Preserve environment variable order in the Pod specification {pull}2341[#2341]
```

alaudazzi commented on docs/release-notes/1.0.0.asciidoc:
```diff
@@ -22,27 +22,42 @@
+* Preserve environment variable order in pod spec {pull}2341[#2341]
+* Reuse the same upscaleState across StatefulSets {pull}2339[#2339] (issue: {issue}2338[#2338])
+* Allow node restart even if cluster health is yellow {pull}2330[#2330]
+* Don&#39;t upgrade pods if empty StatefulSet UpdateRevision {pull}2321[#2321] (issue: {issue}2320[#2320])
```
```suggestion
* Do not upgrade Pods if empty StatefulSet UpdateRevision {pull}2321[#2321] (issue: {issue}2320[#2320])
```

alaudazzi commented on docs/release-notes/1.0.0.asciidoc:
```diff
@@ -22,27 +22,42 @@
+* Allow node restart even if cluster health is yellow {pull}2330[#2330]
+* Don&#39;t upgrade pods if empty StatefulSet UpdateRevision {pull}2321[#2321] (issue: {issue}2320[#2320])
+* Fix how cluster.initial_master_nodes is set {pull}2315[#2315] (issue: {issue}2291[#2291])
+* Wait for webhook key to be present in fs {pull}2312[#2312]
```
Not sure what `fs` is. Shall we spell it?

anyasabo commented on docs/release-notes/highlights-1.0.0.asciidoc:
```diff
@@ -0,0 +1,38 @@
+[id="{p}-multi-namespace"]
+=== Multiple namespace management
+
+ECK can now manage multiple specific namespaces simultaneously (previously only all namespaces or a single namespace was possible).  See <<{p}-operator-config>> for more information.
```
correct, ill reword
